### PR TITLE
Pip8

### PR DESCRIPTION
**Pull request by glogiotatidis**
- Migrate requirements to pip8 hash checking mechanism
- Remove peep
- Add pipstrap
- Upgrade travis and docker tests
- Extra: separated requiments used to test and run the app with the ones
  needed for development only (i.e. building docs, debugging, etc)

	
@peterbe r?

### CODE REVIEW

File: {{ cookiecutter.project_name }}/Dockerfile [hunk 1]
```diff
@@ -18,14 +18,14 @@ RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
 
 WORKDIR /app
 
-# Pin a known to work with peep pip version.
-RUN pip install pip==6.0.0
+# Get pip8
+COPY bin/pipstrap.py bin/pipstrap.py
```
Comment (peterbe) [inline {{ cookiecutter.project_name }}/Dockerfile]: ???

**Comment by peterbe** [comment]
I've grown quite fond of every project having a `requirements.txt`. I understand the hack you did with `-r base.txt` but all that just for the sake of `requirements/docs.txt`? You could just instead create `docs/requirements.txt` which feels more natural to me. 

Also, if someone wants to add another dependency they might be templated to write `hashin somepackage` which will, by default, look up the latest version AND put it into `./requirements.txt`.

**Comment by peterbe** [comment]
We used to have `peepin` [in the requirements.txt](https://github.com/mozilla/sugardough/blob/1a54dc9b7d227b249f5ed3d2f89d59c9186b504a/%7B%7B%20cookiecutter.project_name%20%7D%7D/requirements.txt#L9). Should we include `hashin` instead?

File: {{ cookiecutter.project_name }}/Dockerfile (hunk 1 above)
Comment (peterbe) [inline {{ cookiecutter.project_name }}/Dockerfile]: I'm not very good with Docker but that just looks odd :)

Comment (glogiotatidis) [inline {{ cookiecutter.project_name }}/Dockerfile]: Hehe I get the confusion, here's what's happening:
 - Docker starts building an image from the base defined in `FROM`. The image has nothing else from your code yet
 - You add files into your image with `ADD` and `COPY`.
 - Here we're copying `bin/pipstrap.py` from the source of the host, i.e. the directory you're working on into the docker image.
 - In the next line we run `pipstrap` to get `pip8`

**Comment by glogiotatidis** [comment]
> I've grown quite fond of every project having a requirements.txt. I understand the hack you did with -r base.txt but all that just for the sake of requirements/docs.txt? You could just instead create docs/requirements.txt which feels more natural to me.
> 
> Also, if someone wants to add another dependency they might be templated to write hashin somepackage which will, by default, look up the latest version AND put it into ./requirements.txt.

Makes sense. I started doing the separation trying to separate base, dev and docs but then `dev` is needed in the docker image to be able to run all the tests and ship the same image you've tested against. So `dev` ended up in `base` and then `docs` is the one left out because it's certainly not needed in the docker image and more importantly we don't need to track all the hashes for documentation builders :)

I'll switch to the one you suggest.

**Comment by glogiotatidis** [comment]
> We used to have peepin in the requirements.txt. Should we include hashin instead?

re-added!

**Comment by glogiotatidis** [comment]
@peterbe updated to address your comments

**Comment by peterbe** [comment]
What about

> We used to have peepin in the requirements.txt. Should we include hashin instead?

??

**Comment by glogiotatidis** [comment]
> What about
> 
>     We used to have peepin in the requirements.txt. Should we include hashin instead?
> 
> ??

See https://github.com/mozilla/sugardough/pull/114#issuecomment-178538697

**Comment by peterbe** [comment]
Grr. My bad. I didn't read for it. I searched and I must have typo'ed. It's clearly there. 

r+

**Comment by glogiotatidis** [comment]
woohoo!

**Pull request merged**